src/Simplex/Messaging/Agent.hs

```diff
@@ -402,16 +409,24 @@ deleteConnectionAsync' c@AgentClient {subQ} corrId connId =
     notifyDeleted :: m ()
     notifyDeleted = atomically $ writeTBQueue subQ (corrId, connId, OK)
 
+-- | Add connection to the new receive queue
+switchConnection' :: AgentMonad m => AgentClient -> ConnId -> m ()
+switchConnection' c connId =
+  withStore c (`getConn` connId) >>= \case
+    SomeConn _ DuplexConnection {} -> enqueueCommand c "" connId Nothing $ AInternalCommand ICQSwitch
+    _ -> throwError $ CMD PROHIBITED
+
 newConn :: AgentMonad m => AgentClient -> ConnId -> Bool -> Bool -> SConnectionMode c -> m (ConnId, ConnectionRequestUri c)
 newConn c connId asyncMode enableNtfs cMode =
   getSMPServer c >>= newConnSrv c connId asyncMode enableNtfs cMode
 
 newConnSrv :: AgentMonad m => AgentClient -> ConnId -> Bool -> Bool -> SConnectionMode c -> SMPServer -> m (ConnId, ConnectionRequestUri c)
 newConnSrv c connId asyncMode enableNtfs cMode srv = do
   AgentConfig {smpClientVRange, smpAgentVRange, e2eEncryptVRange} <- asks config
-  (rq, qUri) <- newRcvQueue c srv smpClientVRange
-  connId' <- setUpConn asyncMode rq $ maxVersion smpAgentVRange
-  addSubscription c rq connId'
+  (_rq, qUri) <- newRcvQueue c "" srv smpClientVRange
+  connId' <- setUpConn asyncMode _rq $ maxVersion smpAgentVRange
+  let rq = (_rq :: RcvQueue) {connId = connId'}
```

Comment:
`_rq` won't be spotted if becomes unused.. it's all very close though, just a nitpick
```suggestion
  (rq', qUri) <- newRcvQueue c "" srv smpClientVRange
  connId' <- setUpConn asyncMode rq' $ maxVersion smpAgentVRange
  let rq = (rq' :: RcvQueue) {connId = connId'}
```